### PR TITLE
add a servo control project to test the servos using hotfire configs

### LINKED ISSUE #57: New project to open and close each servos individually
Using this at test site to more easily debug and test. Preferably, it has to use the same config file the hotefire test project is using.

### PR DESCRIPTION
**Pull request by kovalromank**
Added a ServoControl project which implements HotFire's states `ST_INIT`, `ST_WAIT_FOR_INIT`, `ST_WAIT_FOR_FILLING`, `ST_FILLING`, `ST_DONE` and `ST_ABORT_FILLING` to make it easier to test all the servos before the HotFire test.

The implementation of each state was just copied over, I couldn't find a way around just copying the code. This means that any changes to the HotFire state machine should be copied to the ServoControl state machine as well.

The new project just imports HotFire's `config.h` and `GpioConfig.h`. I'm not sure if we want that or if the ServoControl should have it's own configs.

Closes #57

### CODE REVIEW

**Comment by sebastiengrd** [comment]
This state machine and the state machine of hotfire test shouldn't be the same. This one should be more simple, and what the project is supposed to do is open and close each of the 5 valves individually one at a time. It will be used to open/close each valves to test them. But it should open and close them with the configs files of the hotfire test

File: projects/ServoControl/config/UOStateMachine.cpp [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    interface->createNewGpioPwmOutput(PWM1_NAME, PWM1_PIN);
+#endif
+
+#if USE_PWM2 == 1
+    interface->createNewGpioPwmOutput(PWM2_NAME, PWM2_PIN);
+#endif
+
+#if USE_OUT1 == 1
+    interface->createNewGpioOutput(OUT1_NAME, OUT1_PIN);
+#endif
```
Comment (ajayyy) [inline projects/ServoControl/config/UOStateMachine.cpp:41]: Can you indent this?

File: projects/ServoControl/config/UOStateMachine.cpp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    if (enabled)
+    {
+        // Open PWM2 if the third bit is set
+        bool open = eventNbr & PWM2_EVENT_ENABLE_MASK;
+
+        gpioData.pwmOutputMap.insert({PWM2_NAME, open ? PWM2_OPEN : PWM2_CLOSE});
+
+        std::cout << "ServoControlSM::Control PWM2 " << (open ? "OPEN" : "CLOSE") << "\n";
+    }
+#endif
```
Comment (ajayyy) [inline projects/ServoControl/config/UOStateMachine.cpp:141]: Can you indent this

**Review by ajayyy — CHANGES_REQUESTED** [review]

**Review by ajayyy — DISMISSED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -4,6 +4,9 @@ project(RocketCode2020 C CXX)
 
 add_subdirectory(./libraries)
 
+if (DEFINED ENV{SERVO_CONTROL} OR SERVO_CONTROL)
+	add_definitions(-D SERVO_CONTROL)
+endif()
```
Comment (ajayyy) [inline CMakeLists.txt:9]:
```suggestion
if (DEFINED ENV{SERVO_CONTROL} OR SERVO_CONTROL)
	add_definitions(-D SERVO_CONTROL=$ENV{SERVO_CONTROL})
endif()
```

**Review by ajayyy — APPROVED** [review]

**Pull request merged**